### PR TITLE
[CELEBORN-1298] Support Spark2.4 with Scala2.12

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title

### Why are the changes needed?
As title

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
GA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,17 @@ package apache-celeborn-${project.version}-bin.tgz will be generated.
 | Flink 1.17 | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           | &#x274C;          | &#x274C;           | &#x274C;           |
 | Flink 1.18 | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           | &#x274C;          | &#x274C;           | &#x274C;           |
 
+To compile Spark 2.4 with Scala 2.12, please use the following command
```
Comment (FMX) [inline README.md:70]:
```suggestion
To compile the client for Spark 2.4 with Scala 2.12, please use the following command
```

**Review by FMX — APPROVED** [review]
LGTM overall except a nit.

File: README.md [hunk 2]
```diff
@@ -67,6 +67,17 @@ package apache-celeborn-${project.version}-bin.tgz will be generated.
...
+
+#### Scala 2.12.8/2.12.9/2.12.10
```
Comment (SteNicholas) [inline README.md:72]:
```suggestion
- Scala 2.12.8/2.12.9/2.12.10
```

File: README.md [hunk 3]
```diff
@@ -67,6 +67,17 @@ package apache-celeborn-${project.version}-bin.tgz will be generated.
...
+```shell
+./build/make-distribution.sh -DskipTests -Pspark-2.4 -Dscala.version=${scala.version} -Dscala.binary.version=2.12 -Dmaven.plugin.scala.version=3.2.2 -Dmaven.plugin.silencer.version=1.6.0
+```
+#### Scala 2.12.13-2.12.18
```
Comment (SteNicholas) [inline README.md:76]:
```suggestion
- Scala 2.12.13-2.12.18
```

**Review by SteNicholas — APPROVED** [review]

**Comment by waitinfuture** [comment]
Thanks, merging to main(v0.5.0)/branch-0.4(v0.4.1)

**Pull request closed without merging**